### PR TITLE
added and emit IBeforeRenderViewEvent

### PR DESCRIPTION
**Pull request by lferran**
needed in https://github.com/guillotinaweb/guillotina_ratelimit/pull/1

### CODE REVIEW

**Comment by vangheem** [comment]
Can we rename it to `IBeforeRenderView` and move it to here: https://github.com/plone/guillotina/blob/master/guillotina/traversal.py#L253

\cc @bloodbare

File: guillotina/events.py [hunk 1]
```diff
@@ -72,6 +73,11 @@ class ObjectMovedEvent(ObjectLocationEvent):
     """An object has been moved"""
 
 
+@implementer(IBeforeRenderView)
+class BeforeRenderView(ObjectEvent):
```
Comment (vangheem) [inline guillotina/events.py]: Can we rename to `BeforeRenderViewEvent`.

Do not inherit from `ObjectEvent`, it's not an event on an object!

Need to implement `def __init__(self, request)` method

Comment (lferran) [reply guillotina/events.py]: oh ok!

Comment (lferran) [reply guillotina/events.py]: can I pass the view object aswell? `def __init__(self, request, view)`
I'd like to use it to get the view's route name, but not sure if that already comes in the request

Comment (vangheem) [reply guillotina/events.py]: yup!

**Pull request merged**